_inc/client/state/recommendations/reducer.js

```diff
@@ -0,0 +1,294 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { combineReducers } from 'redux';
+import { assign, difference, get, mergeWith, union } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getInitialRecommendationsStep } from '../initial-state/reducer';
+import {
+	JETPACK_RECOMMENDATIONS_DATA_ADD_SELECTED_RECOMMENDATION,
+	JETPACK_RECOMMENDATIONS_DATA_ADD_SKIPPED_RECOMMENDATION,
+	JETPACK_RECOMMENDATIONS_DATA_FETCH,
+	JETPACK_RECOMMENDATIONS_DATA_FETCH_RECEIVE,
+	JETPACK_RECOMMENDATIONS_DATA_FETCH_FAIL,
+	JETPACK_RECOMMENDATIONS_DATA_UPDATE,
+	JETPACK_RECOMMENDATIONS_STEP_FETCH_RECEIVE,
+	JETPACK_RECOMMENDATIONS_STEP_UPDATE,
+	JETPACK_RECOMMENDATIONS_UPSELL_FETCH,
+	JETPACK_RECOMMENDATIONS_UPSELL_FETCH_RECEIVE,
+	JETPACK_RECOMMENDATIONS_UPSELL_FETCH_FAIL,
+} from 'state/action-types';
+import { getRewindStatus } from 'state/rewind';
+import { getSetting } from 'state/settings';
+import { getSitePlan, hasActiveProductPurchase, hasActiveScanPurchase } from 'state/site';
+import { isPluginActive } from 'state/site/plugins';
+
+const mergeArrays = ( x, y ) => {
```

Comment:
Why do we need this function? Is `mergeWith` not enough?

Reply:
`merge` on its own doesn't merge the array properties like I need it to here, so I used `mergeWith` and this customizer function to get the desired behavior. See this jsfiddle for an illustration of the differences between `merge` and `mergeWith`: https://jsfiddle.net/8jzfc0qt/.